gateway/src/apicast/configuration.lua

```diff
@@ -140,21 +142,49 @@ function _M.services_limit()
 end
 
 function _M.filter_services(services, subset)
-  subset = subset and util.to_hash(subset) or _M.services_limit()
-  if not subset or not next(subset) then return services end
+  local selected_services = {}
+  local service_regexp_filter  = env.value("APICAST_SERVICES_FILTER_BY_URL")
+
+  if service_regexp_filter then
+    -- Checking that the regexp sent is correct, if not an empty service list
+    -- will be returned.
+    local _, err = match("", service_regexp_filter)
```

Comment:
We normally use the "o" and "j" options in `ngx.re.match` for performance : https://github.com/openresty/lua-nginx-module#ngxrematch

In this particular case it might not matter much, but I think we should be consistent and try to add it always. One thing that we should do in the future (not in this PR) is to define somewhere a helper method that uses those options and call it from every module so we do not need to remember to add "oj" every time.